evolve_automation.user.js

```diff
@@ -7894,6 +7895,10 @@
         }
     }
 
+    function findRequiredResourceWeight(resource) {
+        return state.unlockedBuildings.find(building => building.cost[resource.id] > resource.currentQuantity)?.weighting ?? 0;
```

Comment:
actually, I'm a bit confused - does this grab the 1st building if finds? what if there are multiple buildings that require alloy and have different weights?

Reply:
Yes, it'll grab the first one. `state.unlockedBuildings` is sorted by weight right after populating, so it supposed to grab the one with highest weight.